gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractRegion.java

```diff
@@ -1789,18 +1789,18 @@ private void setAttributes(RegionAttributes attrs,String regionName, InternalReg
     this.evictionAttributes = new EvictionAttributesImpl((EvictionAttributesImpl)attrs
         .getEvictionAttributes());
     if (this.partitionAttributes != null) {
-      ((PartitionAttributesImpl) this.partitionAttributes).computeLocalMaxMemory();
-      
       if (this.evictionAttributes != null
           && this.evictionAttributes.getAlgorithm().isLRUMemory()
-          && this.partitionAttributes.getLocalMaxMemory() != 0
-          && this.evictionAttributes.getMaximum() != this.partitionAttributes.getLocalMaxMemory()) {
+          && this.partitionAttributes.getLocalMaxMemory() > 0
```

Comment:
We should keep the non-equality check for correct behavior of localMaxMemory and Eviction maxMemory.